core/deployment/src/main/java/io/quarkus/deployment/cmd/RunCommandProcessor.java

```diff
@@ -52,7 +52,7 @@ public void defaultJavaCommand(PackageConfig packageConfig,
         List<String> args = new ArrayList<>();
         args.add(determineJavaPath());
 
-        for (Map.Entry<?, ?> e : System.getProperties().entrySet()) {
+        for (Map.Entry<?, ?> e : System.getProperties().entrySet()) { //TODO: this is almost certainly wrong as it pulls in all the system properties Maven has set
```

Comment:
I guess we could select only those that are found on the command line. There is also some relevant code in DevMojo. E.g. we have `systemProperties` as a Maven plugin goal parameter.

Reply:
https://github.com/quarkusio/quarkus/blob/3.10.0/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java#L1422
and perhaps some of https://github.com/quarkusio/quarkus/blob/3.10.0/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java#L1247-L1284

Reply:
Hm... As this is code that runs in the Quarkus build system, I am wondering if we should use `BuildSystemTargetBuildItem.buildSystemProps`

Reply:
Ah, that should be it.

Reply:
PR updated

Reply:
Just checked, it's project properties, which is good but we probably want a bit more.
For example system properties specified on the command line that could be picked with
```
BootstrapMavenOptions.newInstance().getSystemProperties()
```

Reply:
> `BootstrapMavenOptions.newInstance().getSystemProperties()`

Can I use this inside a build step?

Reply:
Ah, good point, it will rely on a Maven-specific env var to get the command line

Reply:
I guess we could enhance it later

Reply:
So do you propose I add it or not?

Reply:
Now I need to figure out hot to import the gradle plugins into IntelliJ :)

Reply:
I tested the Gradle change as well